handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslContext.java

```diff
@@ -486,6 +485,12 @@ private void destroy() {
 
                 SSLContext.free(ctx);
                 ctx = 0;
+                OpenSslKeyMaterialProvider provider = sessionContext().provider;
```

Comment:
should `OpenSslSessionContext` just have a `destroy()` method on it, and internally it can take care of the `OpenSslKeyMaterialProvider#destroy()`? That way we don't have to expose more internals of the class.